runtime/proto/CorfuProtocol.proto

```diff
@@ -0,0 +1,825 @@
+syntax = "proto3";
+
+package org.corfudb.runtime;
+
+import "google/protobuf/any.proto";
+
+option java_package = "org.corfudb.runtime.protocol.proto";
+
+/**
+ * Common Definitions
+ **/
+
+enum ProtocolVersion {
+  v0 = 0;
+}
```

Comment:
I am not really sure what the best strategy is for dealing with message versioning (due to a breaking change).

- One way would be to incorporate the version into the name -- i. e. GetLayoutRequestV1.
- We could also introduce a name-spacing concept. I know this is possible with gRPC, but it might not be possible (or too cumbersome) with plain Protobuf definitions.